### DISCUSSION TITLE
Dropout should be fixed

### DISCUSSION THREAD

**Discussion by dslisleedh** [google/flax#2539] [Q&A]
Described in PR: https://github.com/google/flax/pull/2510

versions:
flax: 0.6.1
jax: 0.3.16
jaxlib: 0.3.14

Dropout with rate 1. means not to update related layers. So when validate model, dropout with rate 1. must return output calculated by initialized weights not zero matrix. Pytorch's dropout layer works like I said.

OR at least limit rate range in [0, 1) like Tensorflow.

    ValueError: Exception encountered when calling layer "dropout_13" (type Dropout).
    
    `rate` must be a scalar tensor or a float in the range [0, 1). Received: rate=1.0
    
    Call arguments received:
      • inputs=tf.Tensor(shape=(16,), dtype=float32)
      • training=True

And i dont know why PR build test failed even though i've passed flax's test codes.  

    Error: Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_4a010cdc-4ad5-41a6-8a48-8cc4c4f73390/b725b957-48cd-4024-b802-dbe141e836db.tar.gz. return code: 2.

**Comment by cgarciae** [comment]
@dslisleedh I wrongly converted your issue to a discussion, can you post it again?

**Reply by dslisleedh** [reply]
ok :) i'll repost this after fix PR
